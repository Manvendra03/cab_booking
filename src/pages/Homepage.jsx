import React from 'react';
import { AiOutlineCar } from 'react-icons/ai';
import { Link } from 'react-router-dom';

const Homepage = () => (
  <div className="flex flex-col relative px-4 items-center justify-center h-screen bg-lime-900">
    <h1 className=" text-center text-4xl gap-1 flex-wrap justify-center flex items-center leading-10 tracking-widest lg:text-5xl font-bold text-lime-100 mb-4">
      Welcome to
      <span className="flex items-center">
        CarBooky
        <AiOutlineCar className=" text-2xl text-lime-100  " />
      </span>
    </h1>
    <p className="text-lime-200 text-base md:text-xl mb-32 md:mb-12">
      Book a ride in just a few clicks!
    </p>
    <div className="flex flex-col md:flex-row mt-12 md:mt-4 absolute bottom-12 left-0 md:static justify-center items-center px-10 gap-4 w-full md:w-auto">
      <Link
        to="/login"
        className="px-10 py-2 md:py-4  w-full font-semibold text-center text-white transition duration-300 rounded-lg hover:from-lime-600 text-2xl hover:to-pink-600 ease bg-gradient-to-br from-gray-500 to-pink-500 md:w-[250px]"
      >
        Log in
      </Link>
      <Link
        to="/signup"
        className="px-10 py-2 md:py-4 text-2xl w-full font-semibold text-center text-white transition duration-300 rounded-lg hover:from-lime-600 hover:to-pink-600 ease bg-gradient-to-br from-gray-500 to-pink-500 md:w-[250px]"
      >
        Sign up
      </Link>
    </div>
  </div>
);

export default Homepage;
