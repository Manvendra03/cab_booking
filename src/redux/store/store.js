import { configureStore } from '@reduxjs/toolkit';
import userReducer from '../slices/userSlice';
import fetchCabReducer from '../slices/cabSlice';
import reservationReducer from '../slices/reservationSlice';
import menuReducer from '../slices/menuSlice';

const store = configureStore({
  reducer: {
    fetchCab: fetchCabReducer,
    user: userReducer,
    reservation: reservationReducer,
    menu: menuReducer,
  },
});

export default store;
