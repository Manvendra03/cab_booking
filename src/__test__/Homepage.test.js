import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Homepage from '../pages/Homepage';
import store from '../redux/store/store';

describe('Homepage', () => {
  it('renders', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Homepage component and check for login link', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>,
    );
    const link = screen.getByText(/Welcome to CarBooky/i);
    expect(link).toBeInTheDocument();
  });
  test('renders Homepage component and check for login link', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>,
    );
    const link = screen.getByText(/Book a ride in just a few clicks!/i);
    expect(link).toBeInTheDocument();
  });
  test('renders Homepage component and check for login link', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>,
    );
    const link = screen.getByText(/Log in/i);
    expect(link).toBeInTheDocument();
  });
  test('renders Homepage component and checkout for sign up link', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>,
    );
    const link = screen.getByText(/Sign up/i);
    expect(link).toBeInTheDocument();
  });
});
