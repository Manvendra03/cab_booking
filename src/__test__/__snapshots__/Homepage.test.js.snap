// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders 1`] = `
<div
  className="flex flex-col relative px-4 items-center justify-center h-screen bg-lime-900"
>
  <h1
    className=" text-center text-4xl leading-10 tracking-widest lg:text-5xl font-bold text-lime-100 mb-4"
  >
    Welcome to CarBooky
  </h1>
  <p
    className="text-lime-200 text-xl mb-32 md:mb-12"
  >
    Book a ride in just a few clicks!
  </p>
  <div
    className="flex flex-col md:flex-row mt-12 md:mt-4 absolute bottom-12 left-0 md:static justify-center items-center px-10 gap-4 w-full md:w-auto"
  >
    <a
      className="px-10 py-4  w-full font-semibold text-center text-white transition duration-300 rounded-lg hover:from-lime-600 text-2xl hover:to-pink-600 ease bg-gradient-to-br from-gray-500 to-pink-500 md:w-[250px]"
      href="/login"
      onClick={[Function]}
    >
      Log in
    </a>
    <a
      className="px-10 py-4 text-2xl w-full font-semibold text-center text-white transition duration-300 rounded-lg hover:from-lime-600 hover:to-pink-600 ease bg-gradient-to-br from-gray-500 to-pink-500 md:w-[250px]"
      href="/signup"
      onClick={[Function]}
    >
      Sign up
    </a>
  </div>
</div>
`;
